### ISSUE TITLE
Single class inference, class score is still 0.98 but not 1

### ISSUE THREAD

**Issue by alan-shangguan** [ultralytics/yolov5#5357]
Before submitting a bug report, please be aware that your issue **must be reproducible** with all of the following,
otherwise it is non-actionable, and we can not help you:

- **Current repo**: run `git fetch && git status -uno` to check and `git pull` to update repo
- **Common dataset**: coco.yaml or coco128.yaml
- **Common environment**: Colab, Google Cloud, or Docker image. See https://github.com/ultralytics/yolov5#environments

If this is a custom dataset/training question you **must include** your `train*.jpg`, `val*.jpg` and `results.png`
figures, or we can not help you. You can generate these with `utils.plot_results()`.

## To Reproduce (REQUIRED)

Input:

```
Just train a model using,

python train.py --single-cls
```

Output:

```
a model in runs/train/exp/weights/best.pt
```

## Expected behavior

run debug mode

python detect.py --weights runs/train/exp/weights/best.pt

we should see that in utils/general.py line 655, print x[:,5] which is the cls_conf and it should be all 1 in single class training, but now they are tensor([0.99923, 0.99898, 0.99948,  ..., 0.97183, 0.97135, 0.97071], device='cuda:0')

## Environment

- OS: Linux Mint
- GPU: 2080 Ti

## Additional context

this problem would affect the overall confidence of detection

![1](https://user-images.githubusercontent.com/12264695/138989239-f0c8a4f0-52a0-4cc5-a000-199046963bf2.jpg)
n.

**Comment by glenn-jocher** [comment]
@yuanyuangoo yes I see. I'm not sure that it's going to affect model performance much, but if you'd like you can experiment with updating output biases here to better handle single-class configurations. If you come up with a solution that works well, and does not affect multi-class training, then please let us know or submit a PR to help future users. Thank you!

https://github.com/ultralytics/yolov5/blob/a4fece8c1480aed46a38a6344b403d79c81bd751/models/yolo.py#L196-L205

**Comment by glenn-jocher** [comment]
@yuanyuangoo current output is set to 
```
torch.sigmoid(torch.tensor([math.log(0.6 / (1 - 0.99))])) = tensor([0.9836])
```
for nc=1

**Comment by glenn-jocher** [comment]
@yuanyuangoo some quick experiments show we could add another few decimal places to this value without impacting multi-class results much, i.e.:
```python
torch.sigmoid(torch.tensor([math.log(0.6 / (1 - 0.9999))]))
Out[16]: tensor([0.9998])

torch.sigmoid(torch.tensor([math.log(0.6 / (5 - 0.99))]))
Out[17]: tensor([0.1302])
torch.sigmoid(torch.tensor([math.log(0.6 / (5 - 0.9999))]))
Out[18]: tensor([0.1304])
```

Can you want to submit a PR with this change?
```python
torch.sigmoid(torch.tensor([math.log(0.6 / (1 - 0.99999))]))
Out[19]: tensor([1.0000])
```

**Comment by glenn-jocher** [comment]
TODO: single-class bias init update

**Comment by glenn-jocher** [comment]
@yuanyuangoo good news 😃! Your original issue may now be fixed ✅ in PR #5520. 

Since you did not submit a PR I did this myself. Numerical precision is now increased and torch returns 1.0000 probability for single-class datasets with this PR.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by alan-shangguan** [comment]
Thank you so much.
